### ISSUE TITLE
Setup: Jetpack stuck in loop, can't log in Jetpack

### ISSUE THREAD

**Issue by thevirajshelke** [Automattic/jetpack#16853]
So I had jetpack plugin installed on my WordPress setup and due so some connection issue, I reinstalled the plugin. Turns out that after installing the plugin and clicking on set up jetpack we get a screen where we can log in using WordPress or google account. Once you log in with any of these options the setup should ideally proceed but it gets stuck on the same screen. It tries connecting and the same page is shown again. If you try logging in again the same thing happens! So basically you are stuck in the loop.

#### Steps to reproduce the issue

1. Reinstall Jetpack plugin (or maybe freshly install it)
2. Click on Activate
3. Try setting up jetpack
4. Log in to either of the accounts (WordPress or Google Account)

#### What I expected
Jetpack authentication should complete and you should see the jetpack dashboard

#### What happened instead
You see the login screen again and again (I have tried reinstalling it more than 10 times and tried signing in using both WordPress & Google account nothing works!)

**Screenshots**
![Screenshot from 2020-08-15 17-59-16](https://user-images.githubusercontent.com/30586796/90312510-aba7bb80-df22-11ea-9433-c87f72962a64.png)

Jetpack plugin version - 8.8.1
WordPress version - 5.5

**Comment by dereksmart** [comment]
Hi @thevirajshelke thanks for the report. We're not able to reproduce this on our end, so I'm thinking it's something with your site specifically. 

Could you please try to connect again, and watch the browser console for errors? I'd check in both the JS console and the network console, looking for any errors in either. 

I might also try connecting in another browser to rule that out as well. 

If you'd like some extra help debugging this, you can send your site's info to our support here https://jetpack.com/contact-support/

Thanks, let me know how it goes!

**Comment by thevirajshelke** [comment]
@dereksmart Thanks for a quick reply!

I have been using the Jetpack plugin for the past couple of months (since May'20) on the same hosting server and the same site. I was already logged into the plugin and my site was connected. After I got the update to the WordPress 5.5 I tried reinstalling Jetpack and I could see this. I have tried disabling all other plugins and tried logging in but no luck!

Also, my site health has no issues, here is the SS:
![Screenshot from 2020-08-16 14-24-17](https://user-images.githubusercontent.com/30586796/90330665-3a244780-dfcc-11ea-9076-7b392138029f.png)

The points you mentioned for further analysis were helpful. So I tried changing the browser first tried with chrome (84.0.4147.125) and the next with firefox (79.0). No errors on the console as well, none in the network tab as well! Check these SS maybe they might give some insights.

Chrome:
![Screenshot from 2020-08-16 14-32-15](https://user-images.githubusercontent.com/30586796/90330812-570d4a80-dfcd-11ea-9a3c-a17a36beb1f7.png)

![Screenshot from 2020-08-16 14-29-59](https://user-images.githubusercontent.com/30586796/90330774-1281af00-dfcd-11ea-8ddb-8a33b9524124.png)

Firefox:
![Screenshot from 2020-08-16 14-35-27](https://user-images.githubusercontent.com/30586796/90330889-e286db80-dfcd-11ea-938a-1641f6af7e20.png)
![Screenshot from 2020-08-16 14-35-20](https://user-images.githubusercontent.com/30586796/90330892-e4e93580-dfcd-11ea-8163-8c64e582c16d.png)

**Comment by thevirajshelke** [comment]
@dereksmart Seems like a bug to me.

So if I right-click the button `Set up Jetpack` and open in a new tab I am able to log in and authenticate the plugin!

**Comment by jeherve** [comment]
Could you contact us via [this contact form](https://jetpack.com/contact-support/?rel=support) and give us more details about your site? That may help us understand what went wrong, and run more tests with you.

Thank you!

**Comment by jeherve** [comment]
@thevirajshelke Could you email us at support at jetpack.com instead?

**Comment by bdpardue** [comment]
I was having this issue with two new sites I was setting up. I switched from Chrome to Firefox browser and the loop problem disappeared!

**Comment by thevirajshelke** [comment]
@bdpardue for me browser didn't make any difference. 

> @dereksmart Seems like a bug to me.
> 
> So if I right-click the button `Set up Jetpack` and open in a new tab I am able to log in and authenticate the plugin!

This worked for me!

**Comment by thevirajshelke** [comment]
@jeherve I have emailed you guys at support@jetpack.com

**Comment by jeherve** [comment]
Thank you! We'll get back to you soon.

**Comment by destiny911** [comment]
I reinstalled website and ran into the same issue - jetpack being the first plug to re-add. Right clicking and opening in a new tab worked for me - thank you for posting that helpful tid bit. I was pulling out my hair!

**Comment by thevirajshelke** [comment]
@destiny911 yeah, I wasted a lot of time figuring it out as well. Opening in a new tab and authorizing is a quick fix as of now. Seems like the plugin is not able to authorize from within the same page. It gets into some weird loop. Let's see what happens next I am in touch with the dev team through email.

**Comment by hazaveh** [comment]
I've hit the exact same issue on fresh WordPress Version 5.5.3. opening the link in a new tab worked.

**Comment by thevirajshelke** [comment]
@hazaveh I hope they fix this asap.

One more weird thing that I have come across is that the Jetpack connection keeps on failing with the website (server). I am not sure if its a bug or some network issues specific to my server/website. Will dig into more before I open an issue for this one as well.

**Comment by jeherve** [comment]
Closing this issue as the connection flow has been reworked and does not happen within wp-admin anymore.
